File: calc.go
package main

import "fmt"

func main() {
    var num1 float64
    var num2 float64
    var operator string
    var result float64
    fmt.Printf("Enter the First Number: ")
    fmt.Scanln(&num1)
    fmt.Printf("Enter Operator (+,-,*,/): ")
    fmt.Scanln(&operator)
    fmt.Printf("Enter the Second Number: ")
    fmt.Scanln(&num2)
    switch operator {
        case "+":
            result = num1 + num2
            fmt.Printf("%.3f %s %.3f = %.3f", num1, operator, num2, result)
        case "-":
            result = num1 - num2
            fmt.Printf("%.3f %s %.3f = %.3f", num1, operator, num2, result)
        case "*":
            result = num1 * num2
            fmt.Printf("%.3f %s %.3f = %.3f", num1, operator, num2, result)
        case "/":
            if num2 == 0 {
                fmt.Println("Cannot Be Divided to 0")
            }else{
                result = num1 / num2
                fmt.Printf("%.3f %s %.3f = %.3f", num1, operator, num2, result)
            }
        default:
            fmt.Println("Invalid Operator")
    }
}

File: calc.rb
# return the result of the calculator
def calculator(num1, num2, operator)
    case operator
    when "+"
        num1 + num2
    when "-"
        num1 - num2
    when "*"
        num1 * num2
    when "/"
        if num2 == 0
            return "Error: division by zero"
        else
            num1 / num2
        end
    end
end

# input operator function
def input_operator
    puts "Please input operator:"
    operator = gets.chomp
    if operator == "+" || operator == "-" || operator == "*" || operator == "/"
        return operator
    elsif operator == "q"
        return "q"
    else
        puts "Invalid operator"
        input_operator
    end
end


while true
    operator = input_operator
    if operator == "q"
        break
    end
    puts "Enter your first number:"
    num1 = gets.chomp
    puts "Enter your second number:"
    num2 = gets.chomp
    result = calculator(num1.to_f, num2.to_f, operator)
    puts "#{num1} #{operator} #{num2} = #{result}"
end